src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -180,6 +184,9 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count, Tas
 	EState *estate = queryDesc->estate;
 	CmdType operation = queryDesc->operation;
 	MemoryContext oldcontext = NULL;
+	DestReceiver *destination = queryDesc->dest;
+	MaterialState *routerState = (MaterialState *) queryDesc->planstate;
+	bool sendTuples = operation == CMD_SELECT || queryDesc->plannedstmt->hasReturning;
```

Comment:
> > @@ -180,6 +184,9 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count, Tas
> >  	EState *estate = queryDesc->estate;
> >  	CmdType operation = queryDesc->operation;
> >  	MemoryContext oldcontext = NULL;
> > +	DestReceiver *destination = queryDesc->dest;
> > +	MaterialState *routerState = (MaterialState *) queryDesc->planstate;
> > +	bool sendTuples = operation == CMD_SELECT || queryDesc->plannedstmt->hasReturning;
> 
> I thought that the master `receives` the tuples, that's why I suggested `receiveTuples`. But, `sending` is also fine if that's how postgres use.

But it's not about the number of tuples to receive from the workers (we
always receive all of them). It's about how many tuples we send to the client.